google-http-client/src/main/java/com/google/api/client/http/HttpRequest.java

```diff
@@ -883,7 +891,12 @@ public HttpResponse execute() throws IOException {
     Preconditions.checkNotNull(requestMethod);
     Preconditions.checkNotNull(url);
 
+    Span span = tracer
+        .spanBuilder(OpenCensusUtils.SPAN_NAME_HTTP_REQUEST_EXECUTE)
```

Comment:
Shouldn't this use resource name as span name ?

Reply:
This span is meant to identify the underlying http request being made. The caller (e.g. in google-cloud-java or whatever library uses google-http-client) should also have a span that identifies the resource/method being called. This span will be a child of that parent span.

Reply:
The java grpc implementation puts in "Sent.<operationName>" as the span, without any assumption about a parent span.  It would be great if the HTTP implementation could also have a span name that has some context about the call, like the URL

Reply:
@bogdandrutu Is it preferable to have the url as the span name here or as an attribute?

Reply:
Ok, as this is a general-use client that is can be and is used outside the scope of our control, we should not include the URL by default as we can't guarantee that the URL does not contain secrets. We should ensure that our higher level clients set the URL or route as the span name.

Reply:
Is it safe to add an attirbute for the URL?

Reply:
Adding as an attribute solves the cardinality issue, but not hiding PII. Users don't have control over this span, so they wouldn't be able opt out.

Reply:
I think adding it as an attribute is good starting point. If somebody complains about that we can reconsider.